### PR TITLE
Add prerelease version support

### LINKED ISSUE #1: Add pre-release version number support
Add support to manage a `Pre-Release` member in the version number following [semver](https://semver.org/) specifications

### PR DESCRIPTION
**Pull request by sowmiyamuthuraman**
Fixes #1

### CODE REVIEW

**Comment by estebanborai** [comment]
Hey @sowmiyamuthuraman ! Thanks for the PR! I will be checking it out soon!
Thanks for your help indeed, have a nice weekend!

File: semvercomp.go [hunk 1]
```diff
@@ -153,3 +179,37 @@ func isValid(version string) bool {
 	pattern, _ := regexp.Compile(`(v)?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`)
 	return pattern.MatchString(version)
 }
+
+//comparePreRelease compares the prerelease of two versions
+func comparePreRelease(preRelease []string, otherPreRelease []string) int {
```
Comment (estebanborai) [inline semvercomp.go]: *Semver* doesn't specify when a *pre-release* value is newer than the other.
[Semver Pre-Release](https://semver.org/#spec-item-9)

Let's pretend I'm trying to get the newer version between these tags:
- 1.0.0-0.3.7 (A pretty obvious version)
- 1.0.0-x.7.z.92 (A valid version with a non quantifiable *pre-release* value)

How do you find the newest, when this is up to the author of the version tag?

I think we should let the user handle the *pre-release* value validation through the `Version` struct.

**Comment by estebanborai** [comment]
Hey @sowmiyamuthuraman, thanks for your time!
Please, remember to update the README with the features implemented in this branch.

Thanks in advance.

File: semvercomp.go (hunk 1 above)
Comment (sowmiyamuthuraman) [reply semvercomp.go]: so,we are not taking pre-release into account for comparing two versions!

Comment (estebanborai) [reply semvercomp.go]: I don't think so, I think the best approach is to let the user manage the array of *pre-release* version members.

**Comment by sowmiyamuthuraman** [comment]
> Please, remember to update the README with the features implemented in this branch.

Hey @estebanborai X.Y.Z nomenclature is used for the version.which is the right way to include the prerelease support in the readme

**Comment by sowmiyamuthuraman** [comment]
Are we considering the versions which begins with V as a valid one?

**Comment by estebanborai** [comment]
> > Please, remember to update the README with the features implemented in this branch.
> 
> Hey @estebanborai X.Y.Z nomenclature is used for the version.which is the right way to include the prerelease support in the readme

I would add `Pr` as the name for the Pre-Release in the table.
Don't overthink it, it's for example purposes so keep it simple.

What do you think @sowmiyamuthuraman ?

**Comment by estebanborai** [comment]
> are we considering the versions which begins with V as a valid one?

As [semver](https://semver.org/#is-v123-a-semantic-version) docs point out.
> “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control.

I think supporting "v" is a good idea as the "v" character is widely used, even GitHub releases recommends you to use "v" when writting a version tag:

> It’s common practice to prefix your version names with the letter v. Some good tag names might be v1.0 or v2.3.4.

**Comment by sowmiyamuthuraman** [comment]
> > are we considering the versions which begins with V as a valid one?
> 
> As [semver](https://semver.org/#is-v123-a-semantic-version) docs point out.
> 
> > “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control.
> 
> I think supporting "v" is a good idea as the "v" character is widely used, even GitHub releases recommends you to use "v" when writting a version tag:
> 
> > It’s common practice to prefix your version names with the letter v. Some good tag names might be v1.0 or v2.3.4.

@estebanborai the semver regex we are using accepts only the lowercase of 'v'. in some of the unit tests versions with uppercase of 'V' is also considered as valid.

**Comment by estebanborai** [comment]
> > > are we considering the versions which begins with V as a valid one?
> > 
> > 
> > As [semver](https://semver.org/#is-v123-a-semantic-version) docs point out.
> > > “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control.
> > 
> > 
> > I think supporting "v" is a good idea as the "v" character is widely used, even GitHub releases recommends you to use "v" when writting a version tag:
> > > It’s common practice to prefix your version names with the letter v. Some good tag names might be v1.0 or v2.3.4.
> 
> @estebanborai the semver regex we are using accepts only the lowercase of 'v'. in some of the unit tests, versions with uppercase of 'V' is also considered as valid.

Hey @sowmiyamuthuraman !
About the semver regexp, you can check the following:
https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

We could use the regexp that support named groups to solve this issue.

If you want to work on this, feel free to open an issue and work on it!
Remember to set the project to `version/1.0.0` as most of the improvements we are working on are targeted to a `1.0.0` version.

Thanks!

**Pull request merged**